Examples/Tests/PythonWrappers/PICMI_inputs_2d.py

```diff
@@ -0,0 +1,216 @@
+                                     pml_divB_cleaning = 1,
+                                     warpx_psatd_update_with_rho = True)
+
+# Initialize diagnostics
+diag_field_list = ["E", "B"]
+field_diag = picmi.FieldDiagnostic(name = 'diag1',
+                                   grid = grid,
+                                   period = 10,
+                                   write_dir = '.',
+                                   warpx_file_prefix = 'Python_wrappers_plt',
+                                   data_list = diag_field_list)
+
+# Initialize simulation
+sim = picmi.Simulation(solver = solver,
+                       max_steps = max_steps,
+                       verbose = 1,
+                       particle_shape = 'cubic',
+                       warpx_current_deposition_algo = 'direct',
+                       warpx_particle_pusher_algo = 'boris',
+                       warpx_field_gathering_algo = 'energy-conserving',
+                       warpx_use_filter = 1)
+
+# Add diagnostics to simulation
+sim.add_diagnostic(field_diag)
+
+# Write input file to run with compiled version
+sim.write_input_file(file_name = 'inputs_2d')
+
+# Whether to include guard cells in data returned by Python wrappers
+include_ghosts = 1
+
+# Compute min and max of fields data
+def compute_minmax(data):
+    vmin = -max(abs(data.min()), abs(data.max()))
+    vmax =  max(abs(data.min()), abs(data.max()))
+    return (vmin, vmax)
+
+# Plot fields data either in valid domain or in PML
+def plot_data(it, data, pml, title, name):
+    fig, ax = plt.subplots(nrows = 1, ncols = 1, gridspec_kw = dict(wspace = 0.5), figsize = [6,5])
+    cax = make_axes_locatable(ax).append_axes('right', size='5%', pad='5%')
+    lw  = 0.8
+    ls = '-'
+    if (pml):
+        # Draw PMLs and ghost regions
+        ax.axvline(x = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = 0+nxg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = -nxpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axvline(x = nx-nxg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axvline(x = nx+nxpml, linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0       , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = 0+nzg   , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = -nzpml  , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz      , linewidth = lw, color = 'grey' , linestyle = ls)
+        ax.axhline(y = nz-nzg  , linewidth = lw, color = 'black', linestyle = ls)
+        ax.axhline(y = nz+nzpml, linewidth = lw, color = 'grey' , linestyle = ls)
+        # Set extent and sliced data
+        extent = np.array([-nxg-nxpml, nx+nxpml+nxg, -nzg-nzpml, nz+nzpml+nzg])
+        X = data[-nxg-nxpml:nx+nxpml+nxg,-nzg-nzpml:nz+nzpml+nzg].transpose()
```

Comment:
@dpgrote It seems like this slicing on the PML arrays is not working (returns an empty array of size zero). For example, I pass `data = Expml[:,:,0]` to the function `plot_data` and then take this slice here, which doesn't work. However, if I take the same slice directly on `Expml` (without the intermediate variable `data`), then it's fine. Do you happen to know if passing through an intermediate variable changes the lower and upper bounds of these PML arrays somehow?

Reply:
For instance, if I set `data = Expml[:,:,0]`, then I get:
- `data[-18:-1,:].shape` returns `(17, 165)` (this one seems right)
- `data[-18:0,:].shape` returns `(0, 165)` (this one is of size zero)

Reply:
The way negative indices are handled in the wrapper objects is different than with numpy arrays. With the wrapper, the indexing is done the same way that it is done in the C++ code, where negative indices refer to guard cells. The indices are relative to having the index `[0,0]` refer to the bottom corner of the valid cells. When you do a slice of a wrapper object, it returns a numpy array. For numpy arrays however, a negative index is relative to the upper edge of the array. So when you use the index `[-18:0,:]`, it has length zero since for the numpy array `-18` is to the right of '0'.